### ISSUE TITLE
Normalize symbol names

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#22769]
Possibly, it is a good idea to normalize symbol names, because `Symbol('a1')==Symbol('a_1')` and `Symbol('a_1')==Symbol('a_{1}')` are both `False` but their representation in latex is same.

**Comment by oscargus** [comment]
Maybe it is enough to simplify replace `^_{}\` with `` when comparing names?

**Comment by ThePauliPrinciple** [comment]
If this is wanted, it should probably do this on object instantiation somehow, custom `__eq__` (and then also custom `__hash__`) can very easily introduce subtle bugs in caching/hashing that are very hard to debug/test for, especially for something like `Symbol`.

In general, I disagree with using the way the LaTeX printer works to determine how SymPy objects should or should not behave. Someone who doesn't use the LaTeX printer will be confused that `Symbol('a_1')==Symbol('a_{1}')==Symbol('a1')`.

**Comment by oscargus** [comment]
> Someone who doesn't use the LaTeX printer will be confused that `Symbol('a_1')==Symbol('a_{1}')==Symbol('a1')`.

Yes, there are probably possible confusions either way... Note that 
`Symbol('a_1')` and `Symbol('a1')` displays the same for the pretty printer, so although not as bad as for LaTeX there is still some scope of confusion offered.

To me it is not obvious that Symbols should compare identical based on their names to start with, but since that is the case (for good reasons, sort of, I assume simplifying parsing is the main reason) one can at least discuss this extension. I can see the benefit when people get an equation and try to change something without looking at the srepr. 

On the other hand, one can think about how much confusion it will actually create to add it. Will people actually use that similar naming for different variables?

I ping in @oscarbenjamin @smichr and @asmeurer on this one.

**Comment by smichr** [comment]
Call it printing sugar that makes them appear the same. We definitely should not make them compare the same. It is debatable whether it is a good idea to print them the same in pretty and latex printers. Can that interpretation be disabled on the printer?

> without looking at the srepr.

The only case I can think of someone getting tripped up is when looking at a pretty printed result of some function that introduced a numbered symbol and then trying to guess the name of the symbol based on the pretty representation instead of retrieving the symbols and assigning a variable to the one(s) of interest.
